listenbrainz/webserver/views/follow.py

```diff
@@ -39,12 +40,13 @@ def follow(user_list):
         "name": current_user.musicbrainz_id,
         "auth_token": current_user.auth_token,
     }
-    spotify_access_token = db_spotify.get_token_for_user(current_user.id)
+    spotify_data = spotify.get_user_dict(current_user.id)
     props = {
         "user": user_data,
         "mode": "follow",
         "follow_list": follow_list_members,
-        "spotify_access_token": spotify_access_token,
+        "spotify_access_token": spotify_data.get("access_token", ""),
+        "spotify_permission": spotify_data.get("permission", ""),
```

Comment:
At this point, maybe we should consider returning an object for spotify props in all the views?  I.E:
```
"spotify": {
    "access_token": XYZ,
    "permissions":XYZ
}
```

It'll be a bit less polluting and better match the refresh-token response structure.

Reply:
Makes sense, I didn't want to break existing stuff which is why I didn't change it. I'll do it.